DashAI/back/models/scikit_learn/numerical_wrapper_for_text.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+from sklearn.feature_extraction.text import CountVectorizer
+
+from DashAI.back.core.schema_fields import BaseSchema, component_field, int_field
+from DashAI.back.dataloaders.classes.dashai_dataset import DashAIDataset
+from DashAI.back.models.scikit_learn.sklearn_like_model import SklearnLikeModel
+from DashAI.back.models.text_classification_model import TextClassificationModel
+
+
+class NumericalWrapperForTextSchema(BaseSchema):
+    """
+    NumericalWrapperForText is a metamodel that allows text classification using
+    tabular classifiers and a tokenizer.
+    """
+
+    tabular_classifier: component_field(
+        description="Tabular model used as the underlying model"
+        "to generate the text classifier.",
+        parent="TabularClassificationModel",
+    )
+    ngram_min_n: int_field(
+        description="Minimum n_gram to use in the vectorizer.",
+        default=1,
+        ge=1,
+    )
+    ngram_max_n: int_field(
+        description="Maximum n_gram to use in the vectorizer.",
+        default=1,
+        le=1,
+    )
+
+
+class NumericalWrapperForText(TextClassificationModel, SklearnLikeModel):
+    """Text classification meta-model."""
+
+    SCHEMA = NumericalWrapperForTextSchema
+
+    def __init__(self, **kwargs) -> None:
+        self.classifier = kwargs["tabular_classifier"]
+        self.vectorizer = CountVectorizer(
+            ngram_range=(kwargs["ngram_min_n"], kwargs["ngram_max_n"])
+        )
+
+    def get_vectorizer(self, input_column: str, output_column: str):
+        """Vectorize input.
+        The vectorized input is assumed to be an array of numbers,
+        so each number is placed in the input_label_idx column,
+        with idx being the position of the number in the array.
```

Comment:
The docstring is not clear. What is this function for? 
To generate the column that will be used for classification?

Reply:
Fixed